src/test/javascript/spec/component/example-modeling/example-modeling-submission.component.spec.ts

```diff
@@ -0,0 +1,284 @@
+        text: 'FeedbackWithReference',
+        referenceId: 'relationshipId',
+        reference: 'reference',
+        credits: 30,
+        correctionStatus: 'CORRECT',
+    } as Feedback;
+    const mockFeedbackWithoutReference = { text: 'FeedbackWithoutReference', credits: 30, type: FeedbackType.MANUAL_UNREFERENCED } as Feedback;
+    const mockFeedbackInvalid = { text: 'FeedbackInvalid', referenceId: '4', reference: 'reference', correctionStatus: FeedbackCorrectionErrorType.INCORRECT_SCORE };
+
+    beforeEach(() => {
+        route = {
+            snapshot: {
+                paramMap: convertToParamMap({ exerciseId: '22', exampleSubmissionId: '35' }),
+                queryParamMap: convertToParamMap({ readOnly: 0, toComplete: 0 }),
+            },
+        } as ActivatedRoute;
+
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, RouterTestingModule.withRoutes([routes[0]]), MockModule(ArtemisExampleModelingSubmissionRoutingModule)],
+            declarations: [ExampleModelingSubmissionComponent, MockComponent(ModelingEditorComponent), MockPipe(ArtemisTranslatePipe)],
+            providers: [MockProvider(ChangeDetectorRef), { provide: ActivatedRoute, useValue: route }],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(ExampleModelingSubmissionComponent);
+                comp = fixture.componentInstance;
+                debugElement = fixture.debugElement;
+                service = debugElement.injector.get(ExampleSubmissionService);
+                alertService = debugElement.injector.get(AlertService);
+                router = debugElement.injector.get(Router);
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should initialize', () => {
+        // GIVEN
+        jest.spyOn(service, 'get').mockReturnValue(of(new HttpResponse({ body: exampleSubmission })));
+        const exerciseService = debugElement.injector.get(ExerciseService);
+        jest.spyOn(exerciseService, 'find').mockReturnValue(of(new HttpResponse({ body: exercise })));
+
+        // WHEN
+        fixture.detectChanges();
+
+        // THEN
+        expect(comp).toBe(comp);
+    });
+
+    it('should handle a new submission', () => {
+        route.snapshot = { ...route.snapshot, paramMap: convertToParamMap({ exerciseId: '22', exampleSubmissionId: 'new' }) } as ActivatedRouteSnapshot;
+
+        // WHEN
+        fixture.detectChanges();
+
+        // THEN
+        expect(comp.isNewSubmission).toBe(true);
+        expect(comp.exampleSubmission).toEqual(new ExampleSubmission());
+    });
+
+    it('should upsert a new modeling submission', () => {
+        // GIVEN
+        const alertSpy = jest.spyOn(alertService, 'success');
+        const serviceSpy = jest.spyOn(service, 'create').mockImplementation((newExampleSubmission, _) => of(new HttpResponse({ body: newExampleSubmission })));
```

Comment:
Since the callback uses values set before calling the service like `submission` field here, I wanted to include the additions made by the component method too. But I removed `_` parameters from lambdas to remove the client style issues you mentioned. (I initially expected using `_` as the parameter name would prevent the `'_' is defined but never used` errors but it looks like this wasn't the case.)